extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -419,32 +435,29 @@ private void handleHibernateORMWithNoPersistenceXml(
                 }
 
                 // sql-load-script
-                // explicit file or default one
-                String importFile = hibernateConfig.sqlLoadScript.orElse("import.sql"); //default Hibernate ORM file imported
-                boolean isNoFile = "no-file".equalsIgnoreCase(importFile);
+                Optional<String> importFile = getSqlLoadScript();
 
-                if (isNoFile) {
+                if (!importFile.isPresent()) {
                     // explicitly set a no file and ignore all other operations
-                    desc.getProperties().setProperty(AvailableSettings.HBM2DDL_IMPORT_FILES, importFile);
+                    desc.getProperties().setProperty(AvailableSettings.HBM2DDL_IMPORT_FILES, NO_SQL_LOAD_SCRIPT_FILE);
                 } else {
                     Optional<Path> loadScriptPath = Optional
-                            .ofNullable(applicationArchivesBuildItem.getRootArchive().getChildPath(importFile));
+                            .ofNullable(applicationArchivesBuildItem.getRootArchive().getChildPath(importFile.get()));
 
                     // enlist resource if present
                     loadScriptPath
                             .filter(path -> !Files.isDirectory(path))
                             .ifPresent(path -> {
                                 String resourceAsString = root.getArchiveRoot().relativize(loadScriptPath.get()).toString();
                                 resourceProducer.produce(new SubstrateResourceBuildItem(resourceAsString));
-                                desc.getProperties().setProperty(AvailableSettings.HBM2DDL_IMPORT_FILES, importFile);
+                                desc.getProperties().setProperty(AvailableSettings.HBM2DDL_IMPORT_FILES, importFile.get());
                                 desc.getProperties().setProperty(AvailableSettings.HBM2DDL_IMPORT_FILES_SQL_EXTRACTOR,
                                         MultipleLinesSqlCommandExtractor.class.getName());
 
                             });
 
                     //raise exception if explicit file is not present (i.e. not the default)
                     hibernateConfig.sqlLoadScript
-                            .filter(o -> !isNoFile)
```

Comment:
So it's not related to this patch but I really think all this logic could be made simpler without all these lambdas everywhere.

Would you be interested in having a look? If not I will give it a try next week.

Reply:
Sure, I'll try to simplify it. Should I include it in this PR or create another one?

Reply:
I would make it an additional commit of this very PR.

Reply:
My intuition is that good old if logic would be more readable but I might be wrong.

Reply:
You weren't wrong at all :)